operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,54 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSourcePolicy holds cluster-wide information about how to handle registry mirror rules.
+// When multple policies are defined, the outcome of the behavior is defined on each field.
+type ImageContentSourcePolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec ImageContentSourcePolicySpec `json:"spec"`
+}
+
+// ImageContentSourcePolicySpec is the specification of the ImageContentSourcePolicy CRD.
+type ImageContentSourcePolicySpec struct {
+	// repositoryDigestMirrors allows images referenced by image digests in pods to be
+	// pulled from alternative mirrored repository locations. The image pull specification
+	// provided to the pod will be compared to the source locations described in RepositoryDigestMirrors
+	// and the image may be pulled down from any of the repositories in the list instead of the
+	// specified repository allowing administrators to choose a potentially faster mirror.
+	// Only image pull specifications that have an image disgest will have this behavior applied
+	// to them - tags will continue to be pulled from the specified repository in the pull spec.
+	// When multiple policies are defined, any overlaps found will be merged together when the mirror
+	// rules are written to `/etc/containers/registries.conf`. For example, if policy A has sources `a, b, c`
+	// and policy B has sources `c, d, e`. Then the mirror rule written to `registries.conf` will be `a, b, c, d, e`
+	// where the duplicate `c` is removed.
```

Comment:
This comment is fine for now, I'm ok with discussing the policy in a follow up before 4.2 GA but after this feature has been prototyped.